File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    d = dict()
    for keys in keymap:
        for key in keys:
            if key in d:
                d[key] = min(keys.index(key) + 1, d[key])
            else:
                d[key] = keys.index(key) + 1
    
    ans = []
    for target in targets:
        cnt = 0
        for s in target:
            if s not in d:
                ans.append(-1)
                break
            cnt += d[s]
        else:
            ans.append(cnt)
        
    return ans

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.java
import java.util.*;

class Solution {
    public int[] solution(String[] keymap, String[] targets) {
        Map<Character,Integer> press=new HashMap<>();
        
        for(int i=0;i<keymap.length;i++){
            for(int j=0;j<keymap[i].length();j++){
                char c=keymap[i].charAt(j);
                int count=j+1;
                press.put(c,Math.min(press.getOrDefault(c,Integer.MAX_VALUE),count));
                
            }
        }
        
        int[] answer=new int[targets.length];
        
        for(int i=0;i<targets.length;i++){
            int total=0;
            for(char c:targets[i].toCharArray()){
                if(!press.containsKey(c)){
                    total=-1;
                    break;
                }
                total+=press.get(c);
            }
            answer[i]=total;
        }
        
        return answer;
}
}